integration-tests/run-intg-test.py

```diff
@@ -0,0 +1,618 @@
+    """
+
+    dom = minidom.parse(urllib2.urlopen(latest_product_build_artifacts_api + "xml"))
+    main_artifact_elements = dom.getElementsByTagName('mainArtifact')
+    for main_artifact in main_artifact_elements:
+        canonical_name_elements = main_artifact.getElementsByTagName("canonicalName")
+        for canonical_name in canonical_name_elements:
+            if canonical_name.firstChild.nodeValue == dist_name + ".pom":
+                parent_node = main_artifact.parentNode
+                return parent_node.getElementsByTagName("url")[0].firstChild.nodeValue
+    return None
+
+
+def get_latest_stable_dist():
+    """Download the latest stable distribution
+    """
+    build_num_artifact = get_latest_stable_artifacts_api()
+    build_num_artifact = re.sub(r'http.//(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}):(\d{1,5})', "https://wso2.org", build_num_artifact)
+    if build_num_artifact is None:
+        raise Exception("Error occured while getting latest stable build artifact API path")
+    relative_path = get_relative_path_of_dist_storage(build_num_artifact + "api/xml")
+    if relative_path is None:
+        raise Exception("Error occured while getting relative path")
+    dist_downl_url = build_num_artifact + "artifact/" + relative_path
+    urllib2.urlretrieve(dist_downl_url, str(get_product_file_path()))
+    extract_path = get_product_file_path()
+    cp.extract_product(Path(extract_path), workspace)
+    logger.info('downloading the latest stable pack from Jenkins is completed.')
+
+
+def create_output_property_fle():
+    """Create output property file which is used when generating email
+    """
+    output_property_file = open("output.properties", "w+")
+    git_url = git_repo_url + "/tree/" + git_branch
+    output_property_file.write("GIT_LOCATION=%s\r\n" % git_url)
+    output_property_file.write("GIT_REVISION=%s\r\n" % tag_name)
+    output_property_file.close()
+
+
+def replace_file(source, destination):
+    """Replace source file to the destination
+    """
+    logger.info('replacing files from:' + str(source) + "to: " + str(destination))
+    if sys.platform.startswith('win'):
+        source = cp.winapi_path(source)
+        destination = cp.winapi_path(destination)
+    shutil.move(source, destination)
+
+
+def main():
+    try:
+        global logger
+        global dist_name
+        logger = function_logger(logging.DEBUG, logging.DEBUG)
+        if sys.version_info < (3, 6):
+            raise Exception(
+                "To run run-intg-test.py script you must have Python 3.6 or latest. Current version info: " + sys.version_info)
+        read_proprty_files()
+        if not validate_property_readings():
+            raise Exception(
+                "Property file doesn't have mandatory key-value pair. Please verify the content of the property file "
+                "and the format")
+        # clone the repository
+        clone_repo()
+
+        if test_mode == "DEBUG":
```

Comment:
Did we test this with DEBUG mode? Doubt if the paths to custom testng files are correct

Reply:
We did not test the DEBUG mode. Only implemented SNAPSHOT and RELEASE modes.